### DISCUSSION TITLE
Duplicated queries

### DISCUSSION THREAD

**Discussion by pc386** [rebing/graphql-laravel#926] [General]
Hello, I am having trouble with query performance. Using telescope I saw that there are 3505 queries, 3478 of which are duplicated.
This is for one request using the pagination. I followed the doc to the dime, and everything is seemingly configured correctly. 

This is my resolve 
```
public function resolve($root, $args, $context, ResolveInfo $resolveInfo, \Closure $getSelectFields)
    {
        $args['sort'] = $args['sort'] ?? ['name' => 'asc'];
        $args['limit'] = $args['limit'] ?? null;
        $args['page'] = $args['page'] ?? null;

        $fields = $getSelectFields();
        Log::info([$fields]);
        $select = $fields->getSelect();
        $with = $fields->getRelations();
        Log::info([$with]);
        Log::info([$select]);
        $query = MediumEloquent::with($with)->select($select);

        if (!empty($args['id'])) {
            $query = $query->where('id', $args['id']);
        }

        if (!empty($args['filter'])) {
            $args['filter'] = array_filter($args['filter']);
            $queryCreator = new QueryCreator(GraphQL::type('MediaFilterInput'), $args['filter']);
            $query = $queryCreator->query($query);
        }

        $queryCreator = new QueryCreator(GraphQL::type('MediaSortInput'), $args['sort']);
        $query = $queryCreator->query($query);
        
        return $query->paginate($args['limit'], ['*'], 'page', $args['page']);
``` 
The performance is abysmal. Even when I query with a limit of 1, I get 98 queries, 80 of which are duplicated. How can I debug this? I guess the eager loading is not working, as I see that the queries always select for a single id, never multiple. However the selectFields are propagated properly.

**Comment by pc386** [comment]
Nevermind I found the issue

**Reply by mfn** [reply]
feel free to share it with the world!

**Comment by pc386** [comment]
There was a custom resolver implement for one of the fields. In this resolve we where calling findOrFail on an eloquent. This caused the massive amounts of queries. I refactored it and it's blazing fast now
